From 251f4b308e63864fb634edd09ced53561772abf8 Mon Sep 17 00:00:00 2001
From: "Chen, ZhiminX" <zhiminx.chen@intel.com>
Date: Tue, 30 Oct 2018 14:07:37 +0800
Subject: [PATCH 0874/1214] ablbc: remove slcan commands

Userspace script has sent commands to IOC, the code at here
is redundancy.

Tracked-On: PKT-1553
Tracked-On: OAM-69459
Change-Id: I805319ccb9a264ef11d371e270c7e2922b7f0b68
Signed-off-by: Chen, ZhiminX <zhiminx.chen@intel.com>
---
 drivers/staging/android/abl/ablbc.c | 52 -------------------------------------
 1 file changed, 52 deletions(-)

diff --git a/drivers/staging/android/abl/ablbc.c b/drivers/staging/android/abl/ablbc.c
index 86cbd8d..1051973 100644
--- a/drivers/staging/android/abl/ablbc.c
+++ b/drivers/staging/android/abl/ablbc.c
@@ -294,47 +294,6 @@ static int set_reboot_target(const char *name)
 
 static const unsigned int DEFAULT_TARGET_INDEX;
 
-static const char * const cold_reset[] = {
-	"/vendor/bin/cansend",
-	"slcan0",
-	"0000FFFF#05015555555555",
-	NULL};
-static const char * const suppress_heartbeat[] = {
-	"/vendor/bin/cansend",
-	"slcan0",
-	"0000FFFF#01035555555555",
-	NULL};
-static const char * const reboot_request[] = {
-	"/vendor/bin/cansend",
-	"slcan0",
-	"0000FFFF#03015555555555",
-	NULL};
-
-static int execute_slcan_command(const char *cmd[])
-{
-#ifdef CONFIG_SEND_SLCAN_ENABLE
-	struct subprocess_info *sub_info;
-	int ret = -1;
-
-	sub_info = call_usermodehelper_setup((char *)cmd[0],
-		(char **)cmd,(char **) NULL, GFP_KERNEL,
-		(void *)NULL, (void*)NULL, (void*)NULL);
-
-	if (sub_info) {
-		ret = call_usermodehelper_exec(sub_info,
-			UMH_WAIT_PROC);
-		pr_info("Exec cmd=%s ret=%d\n", cmd[0], ret);
-	}
-
-	if (ret)
-		pr_err("Failure on cmd=%s ret=%d\n", cmd[0], ret);
-
-	return ret;
-#else
-	return 0;
-#endif
-}
-
 static int ablbc_reboot_notifier_call(struct notifier_block *notifier,
 				      unsigned long what, void *data)
 {
@@ -350,17 +309,6 @@ static int ablbc_reboot_notifier_call(struct notifier_block *notifier,
 				__func__, ret);
 	}
 
-	ret = execute_slcan_command((const char **)suppress_heartbeat);
-	if (ret)
-		goto done;
-
-	ret = execute_slcan_command((const char **)reboot_request);
-	if (ret)
-		goto done;
-
-	ret = execute_slcan_command((const char **)cold_reset);
-
-done:
 	return NOTIFY_DONE;
 }
 
-- 
2.7.4

